lib/widgets/selection.dart

```diff
@@ -2342,7 +2342,7 @@ class SelectAllSelectionAction<T extends Content> extends StatelessWidget {
       final first = all.first;
       assert(all.every((el) => el.runtimeType == first.runtimeType));
 
-      final selectedOfThisType = controller.data.where((el) => el.runtimeType == entryFactory(first, 0).runtimeType);
+      final selectedOfThisType = controller.data.where((el) => el.data.type == entryFactory(first, 0).data.type);
```

Comment:
This solves the problem, but the same does the change in `selection_entry.dart`

Reply:
Oh, I see, the first entry had a `runtimeType`  of  `SelectionEntry<T>`, because we are returning that in `SelectionEntry.fromContent`. But all other entries had a `runtimeType`  of `SelectionEntry<Content>`, because `_selectionPack` didn't cast the `entry`.

This is indeed very problematic, we should avoid `runtimeType` as much as possible.